macosx/Snes9x/AppDelegate.h

```diff
@@ -20,10 +20,17 @@
 
 #import <Cocoa/Cocoa.h>
 #import <snes9x_framework/snes9x_framework.h>
+#import "S9xPreferences/S9xPreferencesWindowController.h"
+
+static NSWindowFrameAutosaveName const kMainWindowIdentifier = @"s9xMainWindow";
```

Comment:
You typically want to have `extern NSWindowAutosaveName const` declarations in header files, and `NSWindowAutosaveName const` values in implementation files. Including the actual string in the header file creates copies in every file that includes the header, and if it's static, that can cause linking issues if the header file is included in multiple libraries.

So in this case, AppDelegate.h would read

`extern NSWindowFrameAutosaveName const kMainWindowIdentifier;`

And in AppDelegate.m:

`NSWindowFrameAutosaveName const kMainWindowIdentifier = @"s9xMainWindow";`

Reply:
Ahh, didn't know that. I am guessing that's the same reason why a lot of C header files contain:

`#ifndef HEADER
#define HEADER
... declarations ...
#632`

It was split up that way before but I didn't realize why that was the case, so I'll try to revert the strings back the way they were.

Reply:
Yes, it's related to the `#ifndef` `#define` header dance.

`#include` in C and C++ literally just copies and pastes the header in the location where the `#include` line is. The preprocessor does this before compilation starts. This is why there are some cases where the order of `#include`s matters.

The Objective-C `#import` does the `#ifndef` and `#define` automatically, but it still copies and pastes the contents of the header file.